spec.html

```diff
@@ -30780,6 +30780,7 @@
         <emu-alg>
           1. Let _obj_ be ? ToObject(*this* value).
           1. Let _len_ be ? ToLength(? Get(_obj_, `"length"`)).
+          1. If _comparefn_ is not *undefined* and IsCallable(_comparefn_) is *false*, throw a *TypeError* exception.
```

Comment:
It seems SpiderMonkey checks `comparefn` first. Other engines check `ToObject(this)` first.

```
$ eshost -e 'Array.prototype.sort.call(null, null)'
#### JavaScriptCore
TypeError: Array.prototype.sort requires that |this| not be null

#### ChakraCore
TypeError: Array.prototype.sort: argument is not a JavaScript object

#### SpiderMonkey
TypeError: invalid Array.prototype.sort argument

#### V8
TypeError: Array.prototype.sort called on null or undefined

#### V8 Harmony
TypeError: Array.prototype.sort called on null or undefined
```